### PR TITLE
Upgrade mongodb to 4.8.1 (min version supporting mongo 6)

### PR DESCRIPTION
**Pull request by vidueirof**
Tested locally with mongo 4, 5 and 6. Also with Atlas (free tier).

### CODE REVIEW

**Comment by vidueirof** [comment]
Mongo node driver 4.x and superior it's not compatible with node 10. More info: https://www.mongodb.com/docs/drivers/node/current/compatibility/#language-compatibility
Node 20 was released, don't you think it's time to deprecate node 10 support?

**Comment by luislobo** [comment]
I have a PR that updates travis and replaces it with Github actions, using newer mongodb versions. I could add other node versions and MongoDB 6

https://github.com/balderdashy/sails-mongo/pull/495

@mikermcneil What do you think about deprecating non-LTS node versions?

**Comment by vidueirof** [comment]
At this time node 10 was left behind a long time ago. I think it's time to move forward.
Take a look at current supported versions:
schedule

**Comment by luislobo** [comment]
I think we should get my PR approved first, so that we have a working CI, and then you can update it with the newer mongdb versions.

File: lib/private/machines/count-records.js [hunk 1]
```diff
@@ -61,7 +61,8 @@ module.exports = {
     //  ╚═╝╚═╝╩ ╩╩ ╩╚═╝╝╚╝╩╚═╝╩ ╩ ╩ ╚═╝  └┴┘┴ ┴ ┴ ┴  ─┴┘└─┘
     var db = inputs.connection;
     var mongoCollection = db.collection(tableName);
-    mongoCollection.find(mongoWhere).count(function countCb(err, nativeResult) {
+    mongoCollection.countDocuments(mongoWhere, function countCb(err, nativeResult) {
+    //mongoCollection.find(mongoWhere).count(function countCb(err, nativeResult) {
```
Comment (eashaw) [inline lib/private/machines/count-records.js:65]: Can you remove this comment?

File: lib/private/machines/create-each-record.js [hunk 2]
```diff
@@ -105,6 +105,15 @@ module.exports = {
       //  ╠═╝╠╦╝║ ║║  ║╣ ╚═╗╚═╗  │││├─┤ │ │└┐┌┘├┤   ├┬┘├┤ │  │ │├┬┘ │││ └─┐ │
       //  ╩  ╩╚═╚═╝╚═╝╚═╝╚═╝╚═╝  ┘└┘┴ ┴ ┴ ┴ └┘ └─┘  ┴└─└─┘└─┘└─┘┴└──┴┘└─└─┘─┘
       // Process record(s) (mutate in-place) to wash away adapter-specific eccentricities.
+
+      var index = 0;
+
+      _.each(s3q.newRecords, function (record){
+        record.id = nativeResult.insertedIds[index];
+        ++index;
+      });
```
Comment (eashaw) [inline lib/private/machines/create-each-record.js:114]: Is it normally `_id`?

File: lib/private/machines/create-record.js [hunk 3]
```diff
@@ -103,6 +103,9 @@ module.exports = {
       // Otherwise, IWMIH we'll be sending back a record:
       // ============================================
 
+      s3q.newRecord.id = nativeResult.insertedId;
+      nativeResult.ops = [s3q.newRecord];
```
Comment (eashaw) [inline lib/private/machines/create-record.js:107]: Mike: Hmmm, why mutate `nativeResult`?
Should this be `_id`?

**Review by eashaw — CHANGES_REQUESTED** [review]

**Comment by DominusKelvin** [comment]
Hey @vidueirof, I'll be closing this PR as this other [PR](https://github.com/balderdashy/sails-mongo/pull/499) has implemented the upgrade and it has been merged. Thank you for working on this 💙

Also you can check here for the upgrade information. 👇🏾

https://blog.sailscasts.com/sails-mongo-v2-1-0

**Pull request closed without merging**